### PR TITLE
Hide erase button when user hasnt accepted policy in AIChat

### PR DESCRIPTION
**Pull request by nullhook**
Resolves https://github.com/brave/brave-browser/issues/32959

This PR does more dev cleanup. If API has an error and should user input be disabled is now part of context and shared in the main, input box and conversation list component.

Additionally, this disables suggested questions buttons when the api has an error or when api is generating 

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/ai_chat/resources/page/components/conversation_list/index.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ function ConversationList () {
   const handleQuestionSubmit = (question: string) => {
     getPageHandlerInstance().pageHandler.submitHumanConversationEntry(question)
   }
-
```
Comment (petemill) [inline components/ai_chat/resources/page/components/conversation_list/index.tsx:35]: remove log?

File: components/ai_chat/resources/page/components/main/index.tsx [hunk 2]
```diff
@@ -35,12 +34,20 @@ function Main () {
...
       <ConversationList />
     )
 
+    eraseConversationButtonElement = (
+      <Button kind="plain-faint" aria-label="Erase conversation history"
+      title="Erase conversation history" onClick={handleEraseClick}>
+          <Icon name="erase" />
+      </Button>
+    )
+
```
Comment (petemill) [inline components/ai_chat/resources/page/components/main/index.tsx:50]: Doesn't seem like there's a conditional here? We're defining `eraseConversationButtonElement` as `null` and then immediately setting it to an element. Why not just put it in line with the rest of the html?

**Comment by petemill** [comment]
@thypon @fmarier do we know why `needs-security-review` was added to this frontend-only PR?

**Comment by fmarier** [comment]
> @thypon @fmarier do we know why `needs-security-review` was added to this frontend-only PR?

Because it hit a "hotword": https://github.com/brave/security-action/blob/8d5f6c4898ae782f55f9c0d7fa78ed620e120600/action.yml#L28

I filed https://github.com/brave/security-action/issues/327 to make this clearer, but in this particular case maybe the word "policy" is a bit too generic to be worthwhile.

File: components/ai_chat/resources/page/components/main/index.tsx (hunk 2 above)
Comment (nullhook) [reply components/ai_chat/resources/page/components/main/index.tsx]: makes sense, done

**Review by petemill — APPROVED** [review]

**Pull request merged**